### DISCUSSION TITLE
How to increase Mocha's timeout?

### DISCUSSION THREAD

**Discussion by denilsonsa** [open-wc/open-wc#2812] [Q&A] [answered]
From my understanding, `web-test-runner` uses `mocha`, [which has a default timeout of 2000ms for each test](https://github.com/mochajs/mocha/blob/2b03d865eec63d627ff229e07d777f25061260d4/lib/runnable.js#L42).

The recent update from Google Chrome 127 to 128 somehow made many of our tests slower. Thus, as a mitigation (or as a solution), I'd like to [change the default timeout for mocha](https://stackoverflow.com/questions/23492043/change-default-timeout-for-mocha). I have hundreds of tests, manually setting the timeout for each one isn't feasible, and isn't a good solution.

The mocha documentation mentions [--timeout](https://mochajs.org/#-timeout-ms-t-ms) commandline parameter, but I don't think I can set it when running the tests through `web-test-runner`. Also, a [search for `timeout`](https://github.com/open-wc/open-wc/discussions?discussions_q=timeout) yields no results here in this discussion section.

(Sidenote: [`waitUntil` is hard-coded to 1000ms](https://github.com/open-wc/open-wc/blob/480509899d95c9fbc3f6913b1532a8b599df3002/packages/testing-helpers/src/helpers.js#L182), and I see no way to override it globally. But that's a different discussion.)

**Comment by thepassle** [comment, marked as answer]
You can configure it in the config: https://modern-web.dev/docs/test-runner/test-frameworks/mocha/#configuring-mocha-options

`waitUntil` _defaults_ to 1000ms, you can override the option if you want

**Reply by denilsonsa** [reply]
Thank you! That link answers perfectly.

> `waitUntil` defaults to 1000ms, you can override the option if you want

But only per function call, never globally. Right? Or am I missing something?
